46b667b4c970d61fbe33a02be9da48eea8a22be0 ramips: mt7621 use rt2880-clock driver
diff --git a/arch/mips/ralink/mt7621.c b/arch/mips/ralink/mt7621.c
index 801edf9fb..466082aad 100644
--- a/arch/mips/ralink/mt7621.c
+++ b/arch/mips/ralink/mt7621.c
@@ -14,6 +14,7 @@
 #include <linux/clkdev.h>
 #include <linux/clk-provider.h>
 #include <dt-bindings/clock/mt7621-clk.h>
+#include <linux/of.h>
 
 #include <asm/mipsregs.h>
 #include <asm/smp-ops.h>
@@ -27,6 +28,7 @@
 #include <pinmux.h>
 
 #include "common.h"
+#include "rt2880-clk_internal.h"
 
 #define MT7621_GPIO_MODE_UART1		1
 #define MT7621_GPIO_MODE_I2C		2
@@ -145,7 +147,7 @@ void __init ralink_clk_init(void)
 {
 	u32 syscfg, xtal_sel, clkcfg, clk_sel, curclk, ffiv, ffrac;
 	u32 pll, prediv, fbdiv;
-	u32 xtal_clk, cpu_clk, bus_clk;
+	u32 xtal_clk, cpu_clk, bus_clk, sys_clk;
 	const static u32 prediv_tbl[] = {0, 1, 2, 2};
 
 	syscfg = rt_sysc_r32(SYSC_REG_SYSTEM_CONFIG0);
@@ -181,9 +183,11 @@ void __init ralink_clk_init(void)
 
 	cpu_clk = cpu_clk / ffiv * ffrac;
 	bus_clk = cpu_clk / 4;
+	sys_clk = 50 * 1000 * 1000;
 
 	clks[MT7621_CLK_CPU] = mt7621_add_sys_clkdev("cpu", cpu_clk);
 	clks[MT7621_CLK_BUS] = mt7621_add_sys_clkdev("bus", bus_clk);
+	clks[MT7621_CLK_SYS] = mt7621_add_sys_clkdev("sys", sys_clk);
 
 	pr_info("CPU Clock: %dMHz\n", cpu_clk / 1000000);
 	mips_hpt_frequency = cpu_clk / 2;
@@ -194,7 +198,32 @@ static void __init mt7621_clocks_init_dt(struct device_node *np)
 	of_clk_add_provider(np, of_clk_src_onecell_get, &clk_data);
 }
 
-CLK_OF_DECLARE(ar7100, "mediatek,mt7621-pll", mt7621_clocks_init_dt);
+CLK_OF_DECLARE(mt7621, "mediatek,mt7621-pll", mt7621_clocks_init_dt);
+
+
+/*
+ * resources for rt2880-clock
+ */
+
+static const struct gate_clk_desc clk_mt7621[GATE_CLK_NUM] __initconst = {
+	[16] = { .name = "i2c", .parent_name = "sys" },
+	[17] = { .name = "i2s", .parent_name = "sys" },
+	[18] = { .name = "spi", .parent_name = "bus" },
+	[19] = { .name = "uartl", .parent_name = "sys" },
+	[20] = { .name = "uart2", .parent_name = "sys" },
+	[21] = { .name = "uart3", .parent_name = "sys" },
+};
+
+const struct of_device_id of_match_rt2880_clk[] __initconst = {
+	{
+		.compatible = "mediatek,mt7621-clock",
+		.data = clk_mt7621,
+	},
+	{
+		/* sentinel */
+	},
+};
+
 
 void __init ralink_of_remap(void)
 {
diff --git a/include/dt-bindings/clock/mt7621-clk.h b/include/dt-bindings/clock/mt7621-clk.h
index 9f8f244f3..c09ecb3ea 100644
--- a/include/dt-bindings/clock/mt7621-clk.h
+++ b/include/dt-bindings/clock/mt7621-clk.h
@@ -12,7 +12,8 @@
 
 #define MT7621_CLK_CPU		0
 #define MT7621_CLK_BUS		1
+#define MT7621_CLK_SYS		2
 
-#define MT7621_CLK_MAX		2
+#define MT7621_CLK_MAX		3
 
 #endif /* __DT_BINDINGS_MT7621_CLK_H */
